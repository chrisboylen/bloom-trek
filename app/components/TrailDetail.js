import React, { Component } from 'react';
import {
  StyleSheet, Text, Image, Linking, View,
} from 'react-native';
import GestureRecognizer from 'react-native-swipe-gestures';
import PropTypes from 'prop-types';

export default class TrailDetail extends Component {
  onSwipeRight() {
    const { resetSelectedTrail } = this.props;
    resetSelectedTrail();
  }

  render() {
    const { selectedTrail } = this.props;
    const {
      name, difficulty, status, summary, length, trail_url, trail_img_url,
    } = selectedTrail;
    return (
      <GestureRecognizer style={styles.container} onSwipeRight={state => this.onSwipeRight(state)}>
        <View style={styles.imageContainer}>
          <Image
            style={styles.image}
            source={{ uri: trail_img_url }}
          />
        </View>
        <View style={styles.text}>
          <Text style={styles.header} onPress={() => { Linking.openURL(trail_url); }}>
            {`${name}   `}
            <Image style={{ height: 18, width: 18 }} source={require('../../assets/icons/arrow.png')} />
          </Text>
          <Text><Text style={{ fontWeight: 'bold' }}>Difficulty: </Text>{difficulty}</Text>
          <Text><Text style={{ fontWeight: 'bold' }}>Length: </Text>{length} miles</Text>
          <Text><Text style={{ fontWeight: 'bold' }}>Status: </Text>{status}</Text>
          <Text style={{ marginTop: 5 }}>
            {summary}
          </Text>
        </View>
      </GestureRecognizer>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: '85%',
  },
  header: {
    width: '100%',
    marginBottom: 5,
    color: '#005900',
    fontSize: 18,
    fontWeight: 'bold',
  },
  text: {
    height: '35%',
    padding: 10,
  },
  imageContainer: {
    height: '50%',
    width: '100%',
  },
  image: {
    height: '100%',
  },
});


TrailDetail.propTypes = {
  resetSelectedTrail: PropTypes.func,
  selectedTrail: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
      PropTypes.object]),
  ),
};
