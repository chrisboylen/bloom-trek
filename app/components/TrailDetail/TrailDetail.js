import React, { Component } from 'react';
import {
  StyleSheet, Text, Image, Linking,
} from 'react-native';
import GestureRecognizer from 'react-native-swipe-gestures';
import PropTypes from 'prop-types';

export default class TrailDetail extends Component {
  onSwipeRight() {
    const { resetSelectedTrail } = this.props;
    resetSelectedTrail();
  }

  render() {
    const { trailInfo } = this.props;
    const {
      name, difficulty, status, summary, length, trail_url, trail_img_url,
    } = trailInfo;
    return (
      <GestureRecognizer style={styles.container} onSwipeRight={state => this.onSwipeRight(state)}>
        <Text onPress={() => { Linking.openURL(trail_url); }}>{name}</Text>
        <Text>Difficulty: {difficulty}</Text>
        <Text>Length: {length} miles</Text>
        <Text>Status: {status}</Text>
        <Text style={{ marginBottom: 10 }}>
          {summary}
        </Text>
        <Image
          style={styles.image}
          source={{ uri: trail_img_url }}
        />
      </GestureRecognizer>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: 225,
  },
  image: {
    height: 50,
    width: 50,
  },
});


TrailDetail.propTypes = {
  resetSelectedTrail: PropTypes.func,
  trailInfo: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
};
