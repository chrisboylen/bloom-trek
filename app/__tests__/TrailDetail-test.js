import React from 'react';
import { shallow } from 'enzyme';
import TrailDetail from '../components/TrailDetail';

/* global spyOn */

describe('TrailDetail', () => {
  let mockResetSelectedTrail;
  let mockSelectedTrail;
  let wrapper;

  beforeEach(() => {
    mockResetSelectedTrail = jest.fn();
    mockSelectedTrail = {
      id: 1,
      name: 'Castlewood Canyon - Rimrock to Creek Bottom Loop',
      uid: 7006769,
      difficulty: 'medium',
      length: 4.6,
      status: 'All Clear',
      trail_img_url: 'https://cdn-files.apstatic.com/hike/7006325_medium_1438452681.jpg',
      trail_url: 'https://www.hikingproject.com/trail/7006769/castlewood-canyon-rimrock-to-creek-bottom-loop',
      summary: 'A fun, rolling trail with a sense of seclusion, great geology, and historical sites.',
      park_id: 1,
      coords: {
        latitude: 39.3598,
        longitude: -104.7682,
      },
    };
    wrapper = shallow(
      <TrailDetail
        selectedTrail={mockSelectedTrail}
        resetSelectedTrail={mockResetSelectedTrail}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('onSwipeRight', () => {
    it('should invoke resetSelectedTrail when invoked', () => {
      wrapper.instance().onSwipeRight();

      expect(mockResetSelectedTrail).toHaveBeenCalled();
    });
  });

  it('should invoke onSwipeRight on right swipe', () => {
    const spy = spyOn(wrapper.instance(), 'onSwipeRight');
    wrapper.instance().forceUpdate();

    wrapper.find('GestureRecognizer').props().onSwipeRight();

    expect(spy).toHaveBeenCalled();
  });
});
