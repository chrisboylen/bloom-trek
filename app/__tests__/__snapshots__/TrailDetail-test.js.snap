// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrailDetail should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrailDetail
    resetSelectedTrail={[MockFunction]}
    selectedTrail={
      Object {
        "coords": Object {
          "latitude": 39.3598,
          "longitude": -104.7682,
        },
        "difficulty": "medium",
        "id": 1,
        "length": 4.6,
        "name": "Castlewood Canyon - Rimrock to Creek Bottom Loop",
        "park_id": 1,
        "status": "All Clear",
        "summary": "A fun, rolling trail with a sense of seclusion, great geology, and historical sites.",
        "trail_img_url": "https://cdn-files.apstatic.com/hike/7006325_medium_1438452681.jpg",
        "trail_url": "https://www.hikingproject.com/trail/7006769/castlewood-canyon-rimrock-to-creek-bottom-loop",
        "uid": 7006769,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "height": "50%",
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://cdn-files.apstatic.com/hike/7006325_medium_1438452681.jpg",
              }
            }
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </Component>,
        <Component
          style={
            Object {
              "height": "35%",
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Object {
                "color": "#005900",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            Castlewood Canyon - Rimrock to Creek Bottom Loop   
            <Image
              source={1}
              style={
                Object {
                  "height": 18,
                  "width": 18,
                }
              }
            />
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Difficulty: 
            </Text>
            medium
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Length: 
            </Text>
            4.6
             miles
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Status: 
            </Text>
            All Clear
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginTop": 5,
              }
            }
          >
            A fun, rolling trail with a sense of seclusion, great geology, and historical sites.
          </Text>
        </Component>,
      ],
      "onSwipeRight": [Function],
      "style": Object {
        "height": "85%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            source={
              Object {
                "uri": "https://cdn-files.apstatic.com/hike/7006325_medium_1438452681.jpg",
              }
            }
            style={
              Object {
                "height": "100%",
              }
            }
          />,
          "style": Object {
            "height": "50%",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": Object {
              "uri": "https://cdn-files.apstatic.com/hike/7006325_medium_1438452681.jpg",
            },
            "style": Object {
              "height": "100%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Object {
                  "color": "#005900",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                  "width": "100%",
                }
              }
            >
              Castlewood Canyon - Rimrock to Creek Bottom Loop   
              <Image
                source={1}
                style={
                  Object {
                    "height": 18,
                    "width": 18,
                  }
                }
              />
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Difficulty: 
              </Text>
              medium
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Length: 
              </Text>
              4.6
               miles
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Status: 
              </Text>
              All Clear
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginTop": 5,
                }
              }
            >
              A fun, rolling trail with a sense of seclusion, great geology, and historical sites.
            </Text>,
          ],
          "style": Object {
            "height": "35%",
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "Castlewood Canyon - Rimrock to Creek Bottom Loop   ",
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 18,
                      "width": 18,
                    }
                  }
                />,
              ],
              "ellipsizeMode": "tail",
              "onPress": [Function],
              "style": Object {
                "color": "#005900",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 5,
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": Array [
              "Castlewood Canyon - Rimrock to Creek Bottom Loop   ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "source": 1,
                  "style": Object {
                    "height": 18,
                    "width": 18,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Difficulty: 
                </Text>,
                "medium",
              ],
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Difficulty: ",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "fontWeight": "bold",
                  },
                },
                "ref": null,
                "rendered": "Difficulty: ",
                "type": [Function],
              },
              "medium",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Length: 
                </Text>,
                4.6,
                " miles",
              ],
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Length: ",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "fontWeight": "bold",
                  },
                },
                "ref": null,
                "rendered": "Length: ",
                "type": [Function],
              },
              4.6,
              " miles",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Status: 
                </Text>,
                "All Clear",
              ],
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Status: ",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "fontWeight": "bold",
                  },
                },
                "ref": null,
                "rendered": "Status: ",
                "type": [Function],
              },
              "All Clear",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "A fun, rolling trail with a sense of seclusion, great geology, and historical sites.",
              "ellipsizeMode": "tail",
              "style": Object {
                "marginTop": 5,
              },
            },
            "ref": null,
            "rendered": "A fun, rolling trail with a sense of seclusion, great geology, and historical sites.",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "height": "50%",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://cdn-files.apstatic.com/hike/7006325_medium_1438452681.jpg",
                }
              }
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </Component>,
          <Component
            style={
              Object {
                "height": "35%",
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Object {
                  "color": "#005900",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                  "width": "100%",
                }
              }
            >
              Castlewood Canyon - Rimrock to Creek Bottom Loop   
              <Image
                source={1}
                style={
                  Object {
                    "height": 18,
                    "width": 18,
                  }
                }
              />
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Difficulty: 
              </Text>
              medium
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Length: 
              </Text>
              4.6
               miles
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Status: 
              </Text>
              All Clear
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginTop": 5,
                }
              }
            >
              A fun, rolling trail with a sense of seclusion, great geology, and historical sites.
            </Text>
          </Component>,
        ],
        "onSwipeRight": [Function],
        "style": Object {
          "height": "85%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              source={
                Object {
                  "uri": "https://cdn-files.apstatic.com/hike/7006325_medium_1438452681.jpg",
                }
              }
              style={
                Object {
                  "height": "100%",
                }
              }
            />,
            "style": Object {
              "height": "50%",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": Object {
                "uri": "https://cdn-files.apstatic.com/hike/7006325_medium_1438452681.jpg",
              },
              "style": Object {
                "height": "100%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                onPress={[Function]}
                style={
                  Object {
                    "color": "#005900",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 5,
                    "width": "100%",
                  }
                }
              >
                Castlewood Canyon - Rimrock to Creek Bottom Loop   
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 18,
                      "width": 18,
                    }
                  }
                />
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Difficulty: 
                </Text>
                medium
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Length: 
                </Text>
                4.6
                 miles
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Status: 
                </Text>
                All Clear
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "marginTop": 5,
                  }
                }
              >
                A fun, rolling trail with a sense of seclusion, great geology, and historical sites.
              </Text>,
            ],
            "style": Object {
              "height": "35%",
              "padding": 10,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  "Castlewood Canyon - Rimrock to Creek Bottom Loop   ",
                  <Image
                    source={1}
                    style={
                      Object {
                        "height": 18,
                        "width": 18,
                      }
                    }
                  />,
                ],
                "ellipsizeMode": "tail",
                "onPress": [Function],
                "style": Object {
                  "color": "#005900",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                  "width": "100%",
                },
              },
              "ref": null,
              "rendered": Array [
                "Castlewood Canyon - Rimrock to Creek Bottom Loop   ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "source": 1,
                    "style": Object {
                      "height": 18,
                      "width": 18,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Difficulty: 
                  </Text>,
                  "medium",
                ],
                "ellipsizeMode": "tail",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Difficulty: ",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "fontWeight": "bold",
                    },
                  },
                  "ref": null,
                  "rendered": "Difficulty: ",
                  "type": [Function],
                },
                "medium",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Length: 
                  </Text>,
                  4.6,
                  " miles",
                ],
                "ellipsizeMode": "tail",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Length: ",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "fontWeight": "bold",
                    },
                  },
                  "ref": null,
                  "rendered": "Length: ",
                  "type": [Function],
                },
                4.6,
                " miles",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Status: 
                  </Text>,
                  "All Clear",
                ],
                "ellipsizeMode": "tail",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Status: ",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "fontWeight": "bold",
                    },
                  },
                  "ref": null,
                  "rendered": "Status: ",
                  "type": [Function],
                },
                "All Clear",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "A fun, rolling trail with a sense of seclusion, great geology, and historical sites.",
                "ellipsizeMode": "tail",
                "style": Object {
                  "marginTop": 5,
                },
              },
              "ref": null,
              "rendered": "A fun, rolling trail with a sense of seclusion, great geology, and historical sites.",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
